core/src/main/java/org/mule/runtime/core/api/streaming/CursorProviderFactory.java

```diff
@@ -24,15 +25,51 @@
   /**
    * Optionally creates a new {@link CursorProvider} to buffer the given {@code value}.
    * <p>
-   * Implementations might resolve that the given stream is/should not be buffered and thus
-   * it will return the same given stream. In that case, the stream will be unaltered.
+   * Implementations might resolve that the given stream is/should not be buffered and thus it will return the same given stream.
+   * In that case, the stream will be unaltered.
    *
    * @param eventContext the context of the event on which buffering is talking place
    * @param value the stream to be cursored
+   * @param originatingLocation the {@link ComponentLocation} where the cursor was created
    * @return A {@link CursorProvider} or the same given {@code inputStream}
+   *
+   * @since 4.4.0
+   */
+  Object of(EventContext eventContext, T value, ComponentLocation originatingLocation);
+
+  /**
+   * Optionally creates a new {@link CursorProvider} to buffer the given {@code value}.
+   * <p>
+   * Implementations might resolve that the given stream is/should not be buffered and thus it will return the same given stream.
+   * In that case, the stream will be unaltered.
+   *
+   * @param eventContext the context of the event on which buffering is talking place
+   * @param value the stream to be cursored
+   * @return A {@link CursorProvider} or the same given {@code inputStream}
+   *
+   * @deprecated use {@link #of(EventContext, Object, ComponentLocation)} instead.
```

Comment:
indicate from which version on this is deprecated

Reply:
done